controller/src/test/java/io/pravega/controller/server/bucket/WatermarkWorkflowTest.java

```diff
@@ -329,7 +329,7 @@ public void testWatermarkingWorkflow() {
         }).when(clientFactory).createRevisionedStreamClient(anyString(), any(), any());
 
         @Cleanup
-        PeriodicWatermarking periodicWatermarking = new PeriodicWatermarking(streamMetadataStore, bucketStore, sp -> clientFactory, executor);
+        PeriodicWatermarking periodicWatermarking = new PeriodicWatermarking(streamMetadataStore, bucketStore, sp -> clientFactory, executor, null);
```

Comment:
Line 316:  Please close the clientFactory as it can cause a thread leak...

Reply:
Sure, fixed. This was a preexisting test.